### DISCUSSION TITLE
Flink CDC 2.x使用DataStream API全量➕增量读取mysql数据库，能不能加表

### DISCUSSION THREAD

**Discussion by wangliansong** [apache/flink-cdc#807] [Q&A] [answered]
cdc版本： 2.1
使用API：datastream
我是使用的datastream api 全量+增量的读取mysql的数据，第一次启动读取3张表，采用exactly-once语意启动；
后面业务变更，需要再加3张新表接入数据。
第一种方式新建一个新的flink cdc 任务，去读取这3张新表
        这种方式是可以满足需求的
第二种方式修改第一次的flink cdc 任务，把这3张新的加进去
        这种方式，我想实现的效果是，老的3张表，继续读取增量，新的3张表读取全量+增量，就是自动识别这3张新表，并把数据同步过来，因为要exactly-once所以重启方式是从saveopint启动，测试发现，从saveopint启动，无法做到自动识别新表，不知道现在cdc 2.x有没有这样的功能，还是我用的方式不对

**Comment by gtk96** [comment, marked as answer]
你好，目前版本尚未支持向已有 job 增加新表，社区目前已在准备实现此功能，请保持关注。
具体请参考以下issues:
https://github.com/ververica/flink-cdc-connectors/issues/513

**Comment by zjb-it** [comment]
> cdc版本： 2.1 使用API：datastream 我是使用的datastream api 全量+增量的读取mysql的数据，第一次启动读取3张表，采用exactly-once语意启动； 后面业务变更，需要再加3张新表接入数据。 第一种方式新建一个新的flink cdc 任务，去读取这3张新表 这种方式是可以满足需求的 第二种方式修改第一次的flink cdc 任务，把这3张新的加进去 这种方式，我想实现的效果是，老的3张表，继续读取增量，新的3张表读取全量+增量，就是自动识别这3张新表，并把数据同步过来，因为要exactly-once所以重启方式是从saveopint启动，测试发现，从saveopint启动，无法做到自动识别新表，不知道现在cdc 2.x有没有这样的功能，还是我用的方式不对

您好，请教2个问题

1. 请问如何设置saveopint   ,我使用StreamExecutionEnvironment.setDefaultSavepointDirectory()时报错不支持该操作
2. 如何从saveopint重启

**Reply by fuyun2024** [reply]
设置 savePoint 不需要修改代码，只需在实行执行上加上-s savePointDir(savePoint 路径)

**Reply by yangrong688** [reply]
加表功能已经实现了，预计在2.2版本发布，可以用master分支打包进行尝试
需要将 scan.newly-added-table.enabled 设置为 true
具体的可以参考下实现的pr
https://github.com/ververica/flink-cdc-connectors/pull/777/files

![lQLPDhs4ZB0z1YrNA6_NB4Cwgoo6DMnmFkECLRB3TIDaAA_1920_943](https://user-images.githubusercontent.com/70557521/157678814-13a91261-4372-4bd9-98f6-c3f80ba41f55.png)
